### ISSUE TITLE
Noise in the wave channel in some cases (CH3)

### ISSUE THREAD

**Issue by brNX** [MiSTer-devel/Gameboy_MiSTer#102]
Noise in the channel 3 (and probably the remaining noise in the others) is related to the missing filter (capacitor) present in the (per-channel) DAC , you can clearly hear it in this test rom [sndtest.zip](https://github.com/MiSTer-devel/Gameboy_MiSTer/files/5080748/sndtest.zip).

https://gbdev.gg8.se/wiki/articles/Gameboy_sound_hardware#Obscure_Behavior

```
static double capacitor = 0.0;

double high_pass( double in, bool dacs_enabled )
{
     double out = 0.0;
     if ( dacs_enabled )
     {
         out = in - capacitor;
         
         // capacitor slowly charges to 'in' via their difference
         capacitor = in - out * 0.999958; // use 0.998943 for MGB&CGB
     }
     return out;
}
```

> The charge factor can be calculated for any output sampling rate as 0.999958^(4194304/rate). So if you were applying high_pass() at 44100 Hz, you'd use a charge factor of 0.996.

**Comment by maij** [comment]
I have been looking into the Gameboy APU recently, and I have tried various highpass filtering and I don't think that is the solution to the noise on the wave channel. N.B. on a real DMG/CGB there are two identical highpass filters on the left and right output channels; there isn't one filter per DAC.

Currently I don't know what the cause of the noisy wave channel is but I will keep looking.

**Comment by paulb-nl** [comment]
Fixed in release 20230102
